test/integration/package_id/test_config_package_id.py

```diff
@@ -39,3 +40,16 @@ def test_error_config_package_id():
     c.run("create .", assert_error=True)
     assert "ERROR: core.package_id:config_mode defined, " \
            "but error while loading 'config_version.json'" in c.out
+
+@pytest.mark.parametrize("config_version, mode, result", [
+    ("myconfig/1.2.3#rev1:pid1#prev1", "minor_mode", "myconfig/1.2.Z"),
+    ("myconfig/1.2.3#rev1:pid1#prev1", "patch_mode", "myconfig/1.2.3"),
+])
+def test_config_package_id_clear(config_version, mode, result):
```

Comment:
Most important gap. This is ensuring that you can call `clear()` on the config_version, was not tested until now.